Core/Dialog/PSPNpSigninDialog.cpp

```diff
@@ -0,0 +1,349 @@
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#if defined(_WIN32)
+#include "Common/CommonWindows.h"
+#endif
+#include <TimeUtil.h>
+#include "Common/Data/Text/I18n.h"
+#include "Common/Serialize/Serializer.h"
+#include "Common/Serialize/SerializeFuncs.h"
+#include "Core/Config.h"
+#include "Core/MemMapHelpers.h"
+#include "Core/Util/PPGeDraw.h"
+#include "Core/HLE/sceKernelMemory.h"
+#include "Core/HLE/sceCtrl.h"
+#include "Core/HLE/sceUtility.h"
+#include "Core/HLE/sceNet.h"
+#include "Core/HLE/sceNetAdhoc.h"
+#include "Core/HLE/sceNp.h"
+#include "Core/Dialog/PSPNpSigninDialog.h"
+#include "Common/Data/Encoding/Utf8.h"
+#include "Core/Reporting.h"
+
+
+static const float FONT_SCALE = 0.65f;
+
+// Needs testing.
+const static int NP_INIT_DELAY_US = 200000; 
+const static int NP_SHUTDOWN_DELAY_US = 501000; 
+const static int NP_RUNNING_DELAY_US = 1000000; // faked delay to simulate signin process to give chance for players to read the text on the dialog
+
+PSPNpSigninDialog::PSPNpSigninDialog(UtilityDialogType type) : PSPDialog(type) {
+}
+
+PSPNpSigninDialog::~PSPNpSigninDialog() {
+}
+
+int PSPNpSigninDialog::Init(u32 paramAddr) {
+	// Already running
+	if (ReadStatus() != SCE_UTILITY_STATUS_NONE)
+		return SCE_ERROR_UTILITY_INVALID_STATUS;
+
+	requestAddr = paramAddr;
+	int size = Memory::Read_U32(paramAddr);
+	memset(&request, 0, sizeof(request));
+	// Only copy the right size to support different request format
+	Memory::Memcpy(&request, paramAddr, size);
+	
+	WARN_LOG_REPORT_ONCE(PSPNpSigninDialogInit, SCENET, "NpSignin Init Params: %08x, %08x, %08x, %08x", request.npSigninStatus, request.unknown1, request.unknown2, request.unknown3);
+
+	ChangeStatusInit(NP_INIT_DELAY_US);
+
+	// Eat any keys pressed before the dialog inited.
+	UpdateButtons();
+	okButtonImg = ImageID("I_CIRCLE");
+	cancelButtonImg = ImageID("I_CROSS");
+	okButtonFlag = CTRL_CIRCLE;
+	cancelButtonFlag = CTRL_CROSS;
+	if (request.common.buttonSwap == 1)
+	{
+		okButtonImg = ImageID("I_CROSS");
+		cancelButtonImg = ImageID("I_CIRCLE");
+		okButtonFlag = CTRL_CROSS;
+		cancelButtonFlag = CTRL_CIRCLE;
+	}
+
+	//npSigninResult = -1;
+	startTime = (u64)(time_now_d() * 1000000.0);
+	step = 0;
+
+	StartFade(true);
+	return 0;
+}
+
+void PSPNpSigninDialog::DrawBanner() {
+
+	PPGeDrawRect(0, 0, 480, 22, CalcFadedColor(0x65636358));
+
+	PPGeStyle textStyle = FadedStyle(PPGeAlign::BOX_VCENTER, 0.6f);
+	textStyle.hasShadow = false;
+
+	// TODO: Draw a hexagon icon
+	PPGeDrawImage(10, 5, 11.0f, 10.0f, 1, 10, 1, 10, 10, 10, FadedImageStyle());
+	auto di = GetI18NCategory("Dialog");
+	PPGeDrawText(di->T("Sign In"), 31, 10, textStyle);
+}
+
+void PSPNpSigninDialog::DrawIndicator() {
+	// TODO: Draw animated circle as processing indicator
+	PPGeDrawImage(456, 248, 20.0f, 20.0f, 1, 10, 1, 10, 10, 10, FadedImageStyle());
+}
+
+void PSPNpSigninDialog::DrawLogo() {
+	// TODO: Draw DNAS logo
```

Comment:
it's just a stub for now showing an empty box, since there is no logo on the atlas, but we can probably replace the official DNAS logo with OpenDNAS logo